### ISSUE TITLE
Training is resuming mid-epoch

### ISSUE THREAD

**Issue by shackett** [napistu/napistu-torch#93]
"You're resuming from a checkpoint that ended before the epoch ended and your dataloader is not resumable. This can cause unreliable results if further training is done. Consider using an end-of-epoch checkpoint or make your dataloader resumable by implementing the `state_dict` / `load_state_dict` interface"

I'm guessing the actual exception is a symptom rather than the problem

"Training failed with unexpected error: Early stopping conditioned on metric `val_auc` which is not available. Pass in or modify your `EarlyStopping` callback to use any of the following: `lr-AdamW`"

<img width="1140" height="648" alt="Image" src="https://github.com/user-attachments/assets/3bf2ec3c-f749-4d38-bc35-ab8114408761" />
